src/components/utilities.js

```diff
@@ -28,12 +28,34 @@ export const Utilities = {
   },
 
   async getSupportedVersions() {
-    try {
-      const resp = await fetch(process.env.SUPPORTED_VERSIONS_JSON);
-      const data = await resp.json();
-      return { ...data.supportedVersions };
-    } catch (error) {
-      throw error;
-    }
+    const internalAddonVersions = fetch(process.env.SUPPORTED_VERSIONS_JSON)
+        .then(response => response.json())
+        .then(response => response.supportedVersions);
+
+    const externalAddonVersions = fetch("https://kurl-sh.s3.amazonaws.com/external/addon-registry.json")
+        .then(response => response.json());
+
+    return this.mergeAddonVersions(...await Promise.all([internalAddonVersions, externalAddonVersions]));
+  },
+
+  mergeAddonVersions(internalAddonVersions, externalAddonVersions) {
+    const addons = {};
+    Object.keys(externalAddonVersions).forEach(externalAddonName => {
+      const fileName = externalAddonName.slice(0, externalAddonName.length - 7); // trim off .tar.gz
```

Comment:
Can we use ["path" ](https://nodejs.org/api/path.html#pathextnamepath) here instead of "length - 7"?

Reply:
Sadly this is frontend code, I could try to find a polyfil for this, which shouldn't be "hard" but I don't know if the juice is worth the squeeze.  Thoughts?

Reply:
can we do this instead? 
```javascript
const fileName = externalAddonName.replace(/\.tar\.gz/, '')
```
We can avoid the polyfill that way, but there will be a performance penalty with the regex, not sure if it's enough to make a difference here though.